indico/modules/events/sessions/controllers/management/sessions.py

```diff
@@ -0,0 +1,119 @@
+    return tpl.render_session_list(event, get_active_sessions(event), get_colors())
+
+
+class RHSessionsList(RHManageSessionsBase):
+    """Display list of all sessions within the event"""
+
+    def _process(self):
+        return WPManageSessions.render_template('management/session_list.html', self.event_new.as_legacy,
+                                                event=self.event_new, sessions=get_active_sessions(self.event_new),
+                                                default_colors=get_colors())
+
+
+class RHCreateSession(RHManageSessionsBase):
+    """Create a session in the event"""
+
+    def _get_random_color(self):
+        used_colors = {s.colors for s in self.event_new.sessions.filter_by(is_deleted=False)}
+        unused_colors = set(get_colors()) - used_colors
+        return random.choice(tuple(unused_colors) or get_colors())
+
+    def _process(self):
+        form = SessionForm(obj=FormDefaults(colors=self._get_random_color()))
+        if form.validate_on_submit():
+            create_session(self.event_new, form.data)
+            return jsonify_data(html=_render_session_list(self.event_new))
+        return jsonify_template('events/sessions/management/add_session.html', form=form)
+
+
+class RHModifySession(RHManageSessionBase):
+    """Modify a session"""
+
+    def _process(self):
+        form = SessionForm(obj=self.session)
+        if form.validate_on_submit():
+            update_session(self.session, form.data)
+            return jsonify_data(html=_render_session_list(self.event_new))
+        return jsonify_template('events/sessions/management/add_session.html', form=form)
+
+
+class RHDeleteSessions(RHManageSessionsBase):
+    """Remove multiple sessions"""
+
+    def _process(self):
+        session_ids = map(int, request.form.getlist('session_ids'))
+        sessions = self.event_new.sessions.filter(Session.id.in_(session_ids), ~Session.is_deleted)
+        for sess in sessions:
+            delete_session(sess)
+        return jsonify_data(html=_render_session_list(self.event_new))
+
+
+class RHExportSessionsCSV(RHManageSessionsBase):
+    """Export list of sessions to a CSV"""
+
+    def _process(self):
+        pass
+
+
+class RHExportSessionsPDF(RHManageSessionsBase):
+    """Export list of sessions to a PDF"""
+
+    def _process(self):
+        pass
+
+
+class RHSessionREST(RHManageSessionBase):
+    """Perform update or removal of a session"""
+
+    def _process_DELETE(self):
+        delete_session(self.session)
+        return jsonify_data(html=_render_session_list(self.event_new))
+
+    def _process_PATCH(self):
+        data = request.json
+        if data.viewkeys() > {'colors'}:
+            raise BadRequest
+        if 'colors' in data:
+            colors = ColorTuple(**data['colors'])
+            if colors not in get_colors():
+                raise BadRequest
+            data['colors'] = colors
+        update_session(self.session, data)
+        return jsonify_data()
```

Comment:
`return jsonify()` is enough (`jsonify_data` adds `success: True` and flashed messages, but the code calling it currently does not expect or handle flashed messages so returning them (if there were any) would result in them being lost)